### PR TITLE
Fix hard-coded standard for CUDA

### PR DESCRIPTION
**Pull request by adayton1**
- Fixes a few places where -std=c++11 was hard-coded
- Allows for the CUDA standard to be different than the C++ standard

### CODE REVIEW

**Review by kennyweiss — APPROVED** [review]
Thanks @adayton1 !

File: cmake/SetupCompilerOptions.cmake [hunk 1]
```diff
@@ -314,11 +315,15 @@ if (BLT_CXX_STD)
                             "Valid Options are ( c++98, c++11, c++14, c++17, c++20 )")
     endif()
 
+    message(STATUS "Standard C++${CMAKE_CXX_STANDARD} selected")
+
     if (ENABLE_CUDA)
-       set(CMAKE_CUDA_STANDARD ${CMAKE_CXX_STANDARD})
-    endif()
+       if (NOT DEFINED CMAKE_CUDA_STANDARD)
```
Comment (white238) [inline cmake/SetupCompilerOptions.cmake:321]: 👍

File: .gitlab/build_and_test.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ mkdir $BUILD_DIR
 cd $BUILD_DIR
 
 cmake -C ../host-configs/llnl/$SYS_TYPE/$HOST_CONFIG ../tests/internal
-make -j8
+make VERBOSE=1 -j8
```
Comment (adayton1) [inline .gitlab/build_and_test.sh:17]: Does this need to be removed?

Comment (white238) [reply .gitlab/build_and_test.sh:17]: it's good to be able to just look at the log in the CI

File: host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_link_with_nvcc.cmake [hunk 3]
```diff
@@ -58,15 +58,16 @@ set(BLT_OPENMP_LINK_FLAGS "-Xlinker -rpath -Xlinker ${OMP_HOME}/lib -L${OMP_HOME
...
 
-set(CUDA_TOOLKIT_ROOT_DIR "/usr/tce/packages/cuda/cuda-10.1.168" CACHE PATH "")
+set(CUDA_TOOLKIT_ROOT_DIR "/usr/tce/packages/cuda/cuda-11.1.1" CACHE PATH "")
 set(CMAKE_CUDA_COMPILER "${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc" CACHE PATH "")
 set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER} CACHE PATH "")
 
-set (_cuda_arch "sm_70")
-set (CMAKE_CUDA_FLAGS "-restrict -arch ${_cuda_arch} -std=c++11 --expt-extended-lambda -G" CACHE STRING "" )
+set(CMAKE_CUDA_ARCHITECTURES "70" CACHE STRING "")
+set(_cuda_arch "sm_${CMAKE_CUDA_ARCHITECTURES}")
```
Comment (adayton1) [inline host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_link_with_nvcc.cmake:66]: Does this flag need to be set if CMAKE_CUDA_ARCHITECTURES is set?

**Comment by white238** [comment]
@kennyweiss do you mind re-reviewing this.. @adayton1 and I worked on a webex to clean up a bunch of related logic.

File: host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_link_with_nvcc.cmake (hunk 3 above)
Comment (white238) [reply host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_link_with_nvcc.cmake:66]: Maybe a legacy of FindCUDA... todo when we revamp our CUDA logic

File: cmake/SetupCompilerStandards.cmake [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        endif()
+        set(CMAKE_CXX_STANDARD 20)
+    else()
+        message(FATAL_ERROR "${BLT_CXX_STD} is an invalid entry for BLT_CXX_STD. "
+                            "Valid Options are ( c++98, c++11, c++14, c++17, c++20 )")
+    endif()
+
+    message(STATUS "Standard C++${CMAKE_CXX_STANDARD} selected")
+
+    if (ENABLE_CUDA)
```
Comment (kennyweiss) [inline cmake/SetupCompilerStandards.cmake:58]: Should we (alt. do we need to) also handle `hip` here?

File: host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_link_with_nvcc.cmake (hunk 3 above)
Comment (kennyweiss) [reply host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_link_with_nvcc.cmake:66]: Should we add it to an issue? (or start one)?

**Review by kennyweiss — APPROVED** [review]
This is great!  
Thanks @adayton1 and @white238 for iterating and improving this.

File: cmake/SetupCompilerStandards.cmake (hunk 4 above)
Comment (white238) [reply cmake/SetupCompilerStandards.cmake:58]: https://github.com/LLNL/blt/issues/575

**Pull request merged**